controllers/postgres_controller.go

```diff
@@ -716,11 +716,56 @@ func (r *PostgresReconciler) updatePatroniConfig(ctx context.Context, instance *
 		return err
 	}
 	if len(pods.Items) == 0 {
-		r.Log.Info("no leader pod ready, requeuing")
-		// TODO return proper error
-		return errors.New("no leader pods found")
+		r.Log.Info("no leader pod found, selecting all spilo pods as a last resort (might be ok if it is still creating)")
+
+		err = r.updatePatroniConfigOnAllPods(ctx, instance)
+		if err != nil {
+			r.Log.Info("updating patroni config failed, got one or more errors")
+		}
+		return err
 	}
 	podIP := pods.Items[0].Status.PodIP
+
+	return r.httpPatchPatroni(ctx, instance, podIP)
+}
+
+func (r *PostgresReconciler) updatePatroniConfigOnAllPods(ctx context.Context, instance *pg.Postgres) error {
+	pods := &corev1.PodList{}
+	opts := []client.ListOption{
+		client.InNamespace(instance.ToPeripheralResourceNamespace()),
+		client.MatchingLabels{"application": "spilo"},
+	}
+	if err := r.SvcClient.List(ctx, pods, opts...); err != nil {
+		r.Log.Info("could not query pods, requeuing")
+		return err
+	}
+
+	if len(pods.Items) == 0 {
+		r.Log.Info("no spilo pods found at all, requeueing")
+		return errors.New("no spilo pods found at all")
+	}
+
+	// iterate all spilo pods
+	var lastErr error
+	for _, pod := range pods.Items {
+		pod := pod // pin!
+		podIP := pod.Status.PodIP
+		if err := r.httpPatchPatroni(ctx, instance, podIP); err != nil {
+			lastErr = err
+			r.Log.Info("failed to update pod")
```

Comment:
Why not just logging the error here and drop the `lastErr`?

Reply:
Mh. I think having multiple lines with "failed to update pod" and without the error is not very useful, especially when it's logged on other places, too. What about collecting the errors and then log a single error in the end? Also one other question: Why is everything logged on info level without using `Log.Error`? Doesn't this look healthy from the monitoring side then?

Reply:
the error is already logged, as an error, in `r.httpPatchPatroni()`. 

This code is (should be) executed in two different situations:
* Initial creation of standby instances in another k8s cluster (in which case, there is only one pod without the label `spilo-role=standby_leader`), and that pod is starting up and, hopefully, will ultimately accept the HTTP PATCH request. This is normal behaviour.
* Something is completely wrong during the lifetime of the pod, and it lost all of it's labels. As a last resort, select all the running pods and make sure the HTTP PATCH is performed, no matter the state of the pod (although the patroni rest api might be stopped at that point too...). This is not normal then, and the whole purpose of this PR is to remedy that.